Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -1007,17 +1025,22 @@ namespace OpenInfraPlatform {
 				\param	faceVertices3D to be added to the triangulation
 				*/
 
-				std::vector<std::vector<carve::geom::vector<3>>> convertIfcCartesianPoint2DVector(const std::vector<std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>>>& points2D)  const throw(...)
+				std::vector<std::vector<carve::geom::vector<3>>> convertIfcCartesianPoint2DVector(
+					const std::vector<std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>>>& points2D)  const throw(...)
 				{
+					if (points2D.expired()) {
+						throw oip::ReferenceExpiredException(points2D);
+					}
 					std::vector<std::vector<carve::geom::vector<3>>> loop2D = std::vector<std::vector<carve::geom::vector<3>>>();
 					const double lengthFactor = UnitConvert()->getLengthInMeterFactor();
 					const uint32_t numPointsY = points2D.size();
 					loop2D.reserve(numPointsY);
 
-					for (unsigned int j = 0; j < numPointsY; ++j) 
+					for (unsigned int j = 0; j < numPointsY; ++j) //TODO �tefan:

```

Comment:
```suggestion
					for (unsigned int j = 0; j < numPointsY; ++j) //TODO Stefan:
```
Don't use any special character other than normal English alphabet in our code.

Reply:
Resolved: b59a5ad